spec/Section 6 -- Execution.md

```diff
@@ -322,39 +312,315 @@ Unsubscribe(responseStream):
+- If {incrementalDataRecords} is empty, return an unordered map containing
+  {data} and {errors}.
+- Let {incrementalResults} be the result of {YieldIncrementalResults(data,
+  errors, incrementalDataRecords)}.
+- Wait for the first result in {incrementalResults} to be available.
+- Let {initialResult} be that result.
+- Return {initialResult} and {BatchIncrementalResults(incrementalResults)}.
+
+Note: {ExecuteExecutionPlan()} does not directly raise field errors from the
+incremental portion of the Execution Plan.
+
+### Yielding Incremental Results
+
+The procedure for yielding incremental results is specified by the
+{YieldIncrementalResults()} algorithm.
+
+YieldIncrementalResults(data, errors, incrementalDataRecords):
+
+- Let {graph} be the result of {GraphFromRecords(incrementalDataRecords)}.
+- Let {pendingResults} be the result of {GetNonEmptyNewPending(graph)}.
+- Update {graph} to the subgraph rooted at nodes in {pendingResults}.
+- Yield the result of {GetInitialResult(data, errors, pendingResults)}.
+- For each completed child Pending Incremental Data node of a root node in
+  {graph}:
+  - Let {incrementalDataRecord} be the Pending Incremental Data for that node;
+    let {result} be the corresponding completed result.
+  - If {data} on {result} is {null}:
+    - Initialize {completed} to an empty list.
+    - Let {parents} be the parent nodes of {executionGroup}.
+    - Initialize {completed} to an empty list.
+    - For each {pendingResult} of {parents}:
+      - Append {GetCompletedEntry(parent, errors)} to {completed}.
+      - Remove {pendingResult} and all of its descendant nodes from {graph},
+        except for any descendant Incremental Data Record nodes with other
+        parents.
+    - Let {hasNext} be {false}, if {graph} is empty.
+    - Yield an unordered map containing {completed} and {hasNext}.
+    - Continue to the next completed Pending Incremental Data node.
+  - Replace {node} in {graph} with a new node corresponding to the Completed
+    Incremental Data for {result}.
+  - Let {resultIncrementalDataRecords} be {incrementalDataRecords} on {result}.
+  - Update {graph} to {GraphFromRecords(resultIncrementalDataRecords, graph)}.
+  - Let {completedDeferredFragments} be the set of root nodes in {graph} without
+    any child Pending Data nodes.
+  - Let {completedIncrementalDataNodes} be the set of completed Incremental Data
+    nodes that are children of {completedDeferredFragments}.
+  - If {completedIncrementalDataNodes} is empty, continue to the next completed
+    Pending Incremental Data Node.
+  - Initialize {incremental} to an empty list.
+  - For each {node} of {completedIncrementalDataNodes}:
+    - Let {incrementalDataRecord} be the corresponding record for {node}.
+    - Append {GetIncrementalEntry(incrementalDataRecord, graph)} to
+      {incremental}.
+    - Remove {node} from {graph}.
+  - Initialize {completed} to an empty list.
+  - For each {pendingResult} of {completedDeferredFragments}:
```

Comment:
For each {completedSomething} of {completedDeferredFragments}:

Reply:
359441e